pkg/common/connectionstore/connection_lookup.go

```diff
@@ -155,48 +165,99 @@ func (c *ConnectionLookup) GetConnectionRecordByNSThreadID(nsThreadID string) (*
 	if err != nil {
 		return nil, fmt.Errorf("get connectionID by namespaced threadID: %w", err)
 	}
-	// adding prefix for storing connection record
-	k := GetConnectionKeyPrefix()(string(connectionIDBytes))
 
-	connRecordBytes, err := c.transientStore.Get(k)
+	var rec ConnectionRecord
+
+	err = c.get(getConnectionKeyPrefix()(string(connectionIDBytes)), &rec, true)
 	if err != nil {
-		return nil, fmt.Errorf("get connection record by connectionID: %w", err)
+		return nil, fmt.Errorf("faild to get connection record by NS thread ID : %s, cause : %w", nsThreadID, err)
+	}
+
+	return &rec, nil
+}
+
+// GetInvitation finds and parses stored invitation to target type
+func (c *ConnectionLookup) GetInvitation(id string, target interface{}) error {
+	if id == "" {
+		return fmt.Errorf(errMsgInvalidKey)
 	}
 
-	return prepareConnectionRecord(connRecordBytes)
+	return c.get(getInvitationKeyPrefix()(id), target, false)
 }
 
-// Store returns handle to underlying permanent store
-func (c *ConnectionLookup) Store() storage.Store {
-	return c.store
+// GetEventData returns persisted event data for given connection ID
+func (c *ConnectionRecorder) GetEventData(connectionID string) ([]byte, error) {
```

Comment:
the “transient store” seems to be an “event store“ in the connections usage.

Reply:
Yes, recorder maintains different prefix to isolate data

Reply:
We might need a followup issue.

Event data shouldn't be assumed to be transient. A particular configuration may store some or all events to a transient store.

Reply:
Created https://github.com/hyperledger/aries-framework-go/issues/1029
and added TODO